packages/checkout/widgets-lib/src/lib/walletConnect.ts

```diff
@@ -149,4 +159,31 @@ export class WalletConnectManager {
       }
     });
   }
+
+  private async loadWalletListings(): Promise<Response | undefined> {
+    // eslint-disable-next-line max-len
+    const walletListingsApi = `https://explorer-api.walletconnect.com/v3/wallets?projectId=${this.walletConnectConfig.projectId}&ids=${metamaskId},${frontierId},${coinbaseId}`;
+    try {
+      const response = await fetch(walletListingsApi);
+      const data = await response.json();
+      return data;
+    } catch (error) {
+      // eslint-disable-next-line no-console
+      console.error('Error fetching wallet listings', error);
+    }
+    return undefined;
+  }
+
+  public async getWalletLogoUrl(): Promise<string | undefined> {
+    const walletListings = await this.loadWalletListings() as any;
+    const walletName = this.ethereumProvider?.session?.peer.metadata.name;
+    if (!walletListings || !walletName) {
+      return undefined;
```

Comment:
Would do you think about returning a default / fallback URL here? something that we can display if we can't resolve the wallet icon.

Reply:
Good call. I think we should just fallback to the WalletConnect logo in this case

Reply:
@dreamoftrees It currently does fallback to the `WalletConnectSymbol` in the `<Logo>` component if the `getWalletLogoUrl` returns undefined.

Reply:
@sharonsheah perfect sorry missed that!